How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,182 @@
+
+```         
+Runtime Error in model spatial.spatial_stadium (models/spatial/spatial.spatial_stadium.sql) line 9:5: Unknown type: STRING
+```
+
+To remedy this, you can recast the geometry value as a binary points.
+
+```         
+ST_ASBINARY(ST_POINT(stadium.lon, stadium.lat)) AS geometry
+```
+
+-   Once your model is defined correctly, run `dbt build` in your terminal to construct the output in your development database:
+
+```         
+dbt build --select proximity.dist_pin_to_stadium
+```
+
+-   Complete the documentation by configuring it as a [source](https://docs.getdbt.com/docs/build/sources) in the folder's dbt `schema.yml` file, and update the folder's `docs.md` and `columns.md` files.
+
+------------------------------------------------------------------------
+
+## Step 4: Incorporate the Transformed Data into the [Final Model View](https://github.com/ccao-data/data-architecture/tree/master/dbt/models/model).
+
+-   Now that the relevant view(s) have been created, they need to be incorporated in the model view.
+
+-   Identify if the new feature is relevant in the condo and/or the residential model. Then, in `dbt/models/model`, update the relevant `vw_pin_condo_input`, `vw_res_card_input`, or `vw_shared_input` view.
+
+-   If the attribute is spatial, you should also update `crosswalk_year_fill.sql`, `proximity.vw_pin10_proximity_fill.sql`, and `location.vw_pin10_location_fill.sql`. Make sure to fill in missing years of data: If certain years are missing from the data but we think that it did not change during that period, we can forward- or backward-fill it.
+
+------------------------------------------------------------------------
+
+## Step 5: Update the [Model Pipeline](https://github.com/ccao-data/model-res-avm/tree/master) to Use the New Versions of the Model Input Views
+
+-   Create a new branch in the `model-res-avm` repository.
+
+-   Add the new feature to the `predictor` section of `model/models/params.yaml`.
+
+-   If you want others to be able to reproduce your work, you can push the updated model data to DVC using the following commands. If you do this, make sure to set the `run_type` to `test` in `params.yaml`.
+
+-   To test the impact, you can run the full model with the command
+
+```         
+dvc repro -f
+```
+
+-   If you want others to be able to reproduce your work, you can push the updated model data to DVC using the following commands. If you do this, make sure to set the `run_type to `test` in `params.yaml`.
+
+```         
+dvc commit
+dvc push 
+```
+
+------------------------------------------------------------------------
+
+## Troubleshooting
+
+-   Some staff members do not have permission to write to AWS buckets. If you run into an `Access Denied` error at any point, ask senior staff to run your code for you. This is most likely to occur during step 1.
```

Comment:
**[Suggestion, non-blocking]** Based on Michael's recent experience, this will probably also happen during `dvc push` in step 5, right?